modules/EnergyManager/Market.hpp

```diff
@@ -77,7 +76,7 @@ class Market {
 
 private:
     Market* _parent;
-    std::vector<Market> _children;
+    std::vector<std::unique_ptr<Market>> _children;
```

Comment:
I'm not sure why you added `std::unique_ptr` here - I would appreciate if you mention this in the commit message (what was the failure and how it got solved).
I guess, the `Market` objects don't behave well, when the vector gets enhanced or references are taken to these elements and become dangling when the vector gets resized.  If this is the case, a simpler solution would be to use `std::list` instead of `std::vector` in combination with `std::unique_ptr`.  `std::list` also guarantees, that it's elements *stay* at the same address, when other elements get added or removed.

Reply:
Thanks, changed it to `std::list` and added the reasoning in the PR description